delivery_free_fee_removal/models/sale_order.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        super().action_confirm()
+        # Invoice all the free delivery line on order confirmation
+        # Or the order will never be fully invoiced.
+        delivery_lines = self.order_line.filtered(
+            lambda line: line.order_id.state == "sale" and line.is_free_delivery
+        )
+        for line in delivery_lines:
+            line.qty_invoiced = line.product_uom_qty
```

Comment:
I think that you should remove the override that makes the "qty_to_invoice" to 0: https://github.com/OCA/delivery-carrier/blob/8574fee184ab2f01b2df349ed8ab7f8b61777d2f/delivery_free_fee_removal/models/sale_order_line.py#L21-L22

Reply:
Removed